src/modules/dwm.cpp

```diff
@@ -190,6 +191,27 @@ namespace modules {
     m_log.info("%s: Building module", name());
     if (tag == TAG_LABEL_TITLE) {
       m_log.info("%s: Building title", name());
+      if (m_tags_scroll && m_title_label) {
+        bool ignore_empty_tags = m_conf.get<string>(name(), "label-empty", "").empty();
+        auto current_tag_it =
+            std::find_if(m_tags.begin(), m_tags.end(), [](const tag_t& tag) { return tag.state == state_t::FOCUSED; });
+        if (current_tag_it != m_tags.end()) {
+          auto next = current_tag_it + 1 < m_tags.end() ? current_tag_it + 1 : current_tag_it;
+          auto prev = current_tag_it == m_tags.begin() ? current_tag_it : current_tag_it - 1;
+          while (ignore_empty_tags && next->state == state_t::EMPTY && next + 1 < m_tags.end()) {
+            ++next;
+          }
+          while (ignore_empty_tags && prev->state == state_t::EMPTY && prev >= m_tags.begin()) {
+            --prev;
+          }
```

Comment:
I would prefer this logic to be in a separate function, the way I've done for `next_layout()` and `prev_layout`. I think it would help keep the code more readable.